sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -118,7 +120,27 @@ public static PipelineOptions createPipelineOptions(Map<String, String> map) {
     for (Map.Entry<String, String> entry : map.entrySet()) {
       args[i++] = "--" + entry.getKey() + "=" + entry.getValue();
     }
-    PipelineOptions options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+
+    PipelineOptions options;
+    try {
+      options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+    } catch (PipelineOptionUnexpectedPropertyException e) {
+      String errorMessage =
+          String.format(
+              "Pipeline option '%s' is not supported in this context. It must be "
```

Comment:
I use `It must be cleared` here to indicate that `RESET option` is a must-do thing.